src/main/kotlin/se/uulm/snowballr/backend/repository/association/ProjectPaperTableRepo.kt

```diff
@@ -60,15 +88,56 @@ class ProjectPaperTableRepo(
         ResultRow::toProjectPaper,
     )
 
+    /**
+     * Generates the next available local paper ID for the specified project by checking the existing maximum local paper ID
+     * within the project and incrementing it by one. If no local paper IDs exist for the given project, it returns 0.
+     *
+     * @param projectId The unique identifier of the project for which the next local paper ID is to be generated.
+     * @return The next available local paper ID as a [Long].
+     */
+    private suspend fun getNextLocalIdForProject(projectId: UUID): Long = db.query {
+        ProjectPaperTable
+            .selectAll()
+            .where { ProjectPaperTable.projectId eq projectId }
+            .maxOfOrNull { it[ProjectPaperTable.localPaperId] }
+            ?.plus(1)
+            ?: 0L
+    }
+
     override suspend fun getProjectPaperById(id: UUID): ProjectPaper = db.query {
         getProjectPaperByIdOrNull(id) ?: throw NotFoundException(EntityType.PROJECT_PAPER, id.toString())
     }
 
+    override suspend fun doesProjectPaperExistsByPaperId(paperId: UUID): Boolean = db.query {
+        ProjectPaperTable
+            .selectAll()
+            .where { ProjectPaperTable.paperId eq paperId }
+            .empty()
+            .not()
+    }
+
     override suspend fun getAllProjectPapersWithPapers(projectId: UUID): List<ProjectPaperWithPaper> = db.query {
         ProjectPaperTable
             .join(PaperTable, JoinType.INNER, onColumn = ProjectPaperTable.paperId, otherColumn = PaperTable.id)
             .selectAll()
             .where { ProjectPaperTable.projectId eq projectId }
             .map { it.toProjectPaperWithPaper() }
     }
+
+    override suspend fun addPaperToProject(request: GrpcProjectPaper.Add, userId: UUID): ProjectPaper = db.query {
+        val userEntityId = getUserEntityId(userId)
+        val paperId = UUID.fromString(request.paperId)
+        val projectId = UUID.fromString(request.projectId)
```

Comment:
Use the `parseUUID` helper here to handle errors.
```suggestion
        val paperId = parseUUID(request.paperId, EntityType.PAPER)
        val projectId = parseUUID(request.projectId, EntityType.PROJECT)
```